bokeh/application/application.py

```diff
@@ -100,9 +101,26 @@ class Application(object):
     # will function without bringing in tornado.
     _is_a_bokeh_application_class = True
 
-    def __init__(self, *handlers):
+    def __init__(self, *handlers, **kwargs):
+        '''
+        Application factory.
+
+        Args:
+            handlers (`bokeh.server.views.doc_handler.DocHandler`):
+                List of handlers called.  The url is taken from the first one
+                only.
+
+            metadata (dict):
+                May be requested by http://applicationurl/metadata as a json
+                blob.
```

Comment:
Just as a note: it looks like Sphinx doesn't currently show the __init__ documentation, so I'm going to move the docstring to the class docs.